### PR TITLE
fix: compilation bugs

### PR DESCRIPTION
**Pull request by xhebox**

### CODE REVIEW

File: lib/Dialect/Tpu/Interfaces/BM1684X/Store.cpp [hunk 1]
```diff
@@ -41,12 +41,12 @@ void tpu::StoreOp::codegen_local_bm1684x(int64_t n_step, int64_t h_step,
     gdma_format = BM168x::GDMA_VALUE_FORMAT_INT8;
     data_type = DTYPE_INT8;
     if (gi.h_slice == H) {
-      if (W * H & 0x1 == 1) {
```
Comment (xhebox) [inline lib/Dialect/Tpu/Interfaces/BM1684X/Store.cpp:44]: Very intresting that it is evaluated as `(W *H) & (0x1 == 1)` but the result is same as `(W * H & 0x1) == 1`.

File: lib/Dialect/Tpu/Interfaces/Common/Csc.cpp [hunk 2]
```diff
@@ -139,7 +139,7 @@ inline int copy_offset(const int *indices, int32_t * stride) {
 }
 
 void crop(float *input, float *output, const int64_t *input_shape,
-          long int *output_shape, int cur_dim, int *offsets, int *indices) {
```
Comment (xhebox) [inline lib/Dialect/Tpu/Interfaces/Common/Csc.cpp:142]: Lots of type mismatch.

File: lib/Dialect/Tpu/Interfaces/Common/Preprocess.cpp [hunk 3]
```diff
@@ -163,7 +163,7 @@ class ReplacePreprocess {
         auto cali_type = module::getCalibratedType(op.getOutput());
         int64_t zeropoint = 0;
         module::getScaleAndZeroPoint(op.getOutput(), qscale, zeropoint,
-                                     this->sign, module::isAsymmetric);
```
Comment (xhebox) [inline lib/Dialect/Tpu/Interfaces/Common/Preprocess.cpp:166]: `isAsymmetric` is a function.

File: lib/Dialect/Tpu/Transforms/BM168x/DynamicNetIr.cpp [hunk 4]
```diff
@@ -567,17 +567,17 @@ void SubnetIr::generate_group_time_step_ir(Operation *op) {
 
     // get and push fw_timestep_base_info_t
     fw_timestep_base_info_t timestep_base_info = {
-      ts_num_and_split_tensor_num : (timestep_num << 16),
```
Comment (xhebox) [inline lib/Dialect/Tpu/Transforms/BM168x/DynamicNetIr.cpp:570]: Really old GNU specific behavior.

**Review by charlesxzb — APPROVED** [review]

**Pull request merged**